### DISCUSSION TITLE
Request: Guidance on WebSocket Implementation in Astro with Separate Logic File

### DISCUSSION THREAD

**Discussion by Trongar** [lilnasy/astro-websockets-demo#2] [Q&A]
I hope you're doing well. I'm working with Astro and aiming to integrate WebSocket functionality into my project. While I've managed a basic implementation, I'm struggling to keep the logic organized within my codebase.

Is there a way to separate WebSocket logic into its own file or module within Astro? I've searched for resources but haven't found clear guidance.

Could you provide insights, examples, or pointers to relevant documentation on this topic?

Thanks for reading

**Comment by lilnasy** [comment]
The example in the repo puts the websocket logic in the astro file for simplicity. For more complex communication, I would put it in its own endpoint file.

For example, `pages/websocket/[roomname].ts` would contain the logic for accepting the websocket upgrade request, along with the more complex code for communication. The browser script here would need to be updated to attempt a websocket connection against `/websocket/roomname` instead of `/roomname`.

https://github.com/lilnasy/astro-websockets-demo/blob/f8385850569cc1be462ad6f7b267d7ec4e14eb01/src/pages/%5B...room%5D.astro#L37
